src/components/UserManagedList.js

```diff
@@ -0,0 +1,271 @@
+      activeFilters: {},
+      activeSort: null,
+      sequence: 0,
+      showFullDetail: false,
+      path: null
+    }
+    this.readOnly = false
+    this.onAddComponent = this.onAddComponent.bind(this)
+    this.onRemoveComponent = this.onRemoveComponent.bind(this)
+    this.loadComponentList = this.loadComponentList.bind(this)
+    this.onRemoveAll = this.onRemoveAll.bind(this)
+    this.showVersionSelectorPopup = this.showVersionSelectorPopup.bind(this)
+    this.applySelectedVersions = this.applySelectedVersions.bind(this)
+    this.doSave = this.doSave.bind(this)
+    this.doSaveAsUrl = this.doSaveAsUrl.bind(this)
+    this.createGist = this.createGist.bind(this)
+    this.saveSpec = this.saveSpec.bind(this)
+  }
+
+  onRemoveComponent(component) {
+    this.props.dispatch(this.updateList({ remove: component }))
+  }
+
+  onRemoveAll() {
+    this.props.dispatch(this.updateList({ removeAll: {} }))
+  }
+
+  showVersionSelectorPopup(component, multiple) {
+    this.setState({ showVersionSelectorPopup: true, multipleVersionSelection: multiple, selectedComponent: component })
+  }
+
+  applySelectedVersions(versions) {
+    const { multipleVersionSelection, selectedComponent } = this.state
+    if (!multipleVersionSelection) {
+      return this.setState({ showVersionSelectorPopup: false }, async () => {
+        if (selectedComponent.changes) {
+          const key = `open${Date.now()}`
+          notification.open({
+            message: 'This definition has some changes. All the unsaved changes will be lost.',
+            btn: (
+              <NotificationButtons
+                onClick={async () => {
+                  await this.onRemoveComponent(selectedComponent)
+                  await this.onAddComponent(
+                    EntitySpec.fromCoordinates({ ...selectedComponent, revision: versions, changes: {} })
+                  )
+                  notification.close(key)
+                }}
+                onClose={() => notification.close(key)}
+                confirmText="Ok"
+                dismissText="Cancel"
+              />
+            ),
+            key,
+            onClose: notification.close(key),
+            duration: 0
+          })
+        } else {
+          await this.onRemoveComponent(selectedComponent)
+          await this.onAddComponent(EntitySpec.fromCoordinates({ ...selectedComponent, revision: versions }))
+        }
+      })
+    }
+    this.setState({ showVersionSelectorPopup: false }, () =>
+      versions.map(version =>
+        this.onAddComponent(EntitySpec.fromCoordinates({ ...selectedComponent, revision: version }))
+      )
+    )
+  }
+
+  revertAll() {
+    this.revert(null, 'Are you sure to revert all the unsaved changes from all the active definitions?')
+  }
+
+  revertDefinition(definition, value) {
+    this.revert(definition, 'Are you sure to revert all the unsaved changes from the selected definition?', value)
+  }
+
+  revert(definition, description, value) {
```

Comment:
suspect that this is (or is largely) duplicate of superclass method. in fact, all the revert logic should be in the superclass as that applies to both user and system managed lists

Reply:
You are right, i can move all of the rever methods into the System Managed List